kv/kvserver: TestTransferLeaseToLaggingNode failed

[(kv/kvserver).TestTransferLeaseToLaggingNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2244163&tab=buildLog) on [master@11a82264044ed8edc7d1fca2b912296d6f576a1a](https://github.com/cockroachdb/cockroach/commits/11a82264044ed8edc7d1fca2b912296d6f576a1a):

```
  github.com/cockroachdb/cockroach/pkg/kv.RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_lookup.go:197 +0xa56
  github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:442 +0x11d
  github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*RangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/range_cache.go:804 +0x449
  github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*RangeDescriptorCache).tryLookup.func1.1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/range_cache.go:658 +0xb1
  github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:135 +0xdd
  github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*RangeDescriptorCache).tryLookup.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/range_cache.go:655 +0x364
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:326 +0x150
  github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*RangeDescriptorCache).tryLookup.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/range_cache.go:643 +0x146
  github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:128 +0x4c

Goroutine 964 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:215 +0xc3
  github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*NodeLiveness).StartHeartbeat()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/node_liveness.go:567 +0x318
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1559 +0x4459
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1474 +0x3649
  github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:427 +0x32a
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerRaw()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:265 +0x199
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:233 +0x73
  github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).doAddServer()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:355 +0x1e4
  github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:226 +0xcdb
  github.com/cockroachdb/cockroach/pkg/testutils/testcluster.StartTestCluster()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:226 +0xcdb
  github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestTransferLeaseToLaggingNode()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:524 +0x804
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:909 +0x199
==================
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode375996274
--- FAIL: TestTransferLeaseToLaggingNode (126.24s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode375996274
    test_log_scope.go:63: use -show-logs to present logs inline
    testing.go:853: race detected during execution of test

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestTransferLeaseToLaggingNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTransferLeaseToLaggingNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>


Jira issue: CRDB-3825

[(kv/kvserver).TestTransferLeaseToLaggingNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2347168&tab=buildLog) on [master@ce41106ed77a8ae0b82c472a81d71d6ccaac7f27](https://github.com/cockroachdb/cockroach/commits/ce41106ed77a8ae0b82c472a81d71d6ccaac7f27):

```
=== RUN   TestTransferLeaseToLaggingNode
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode586401076
--- FAIL: TestTransferLeaseToLaggingNode (78.97s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode586401076
    test_log_scope.go:63: use -show-logs to present logs inline
    replicate_queue_test.go:596: condition failed to evaluate within 45s: remote is not making progress: 0
        goroutine 894273 [running]:
        runtime/debug.Stack(0xc013af1228, 0x92acfe0, 0xc01e828640)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xab
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x94c2c60, 0xc02787da00, 0xc013af1228)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x87
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestTransferLeaseToLaggingNode(0xc02787da00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:596 +0x1531
        testing.tRunner(0xc02787da00, 0x78a5448)
        	/usr/local/go/src/testing/testing.go:909 +0x19a
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:960 +0x652

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTransferLeaseToLaggingNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTransferLeaseToLaggingNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestTransferLeaseToLaggingNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2355430&tab=buildLog) on [master@e824da5a2b33168fa4ef93c83295c8205acfadb7](https://github.com/cockroachdb/cockroach/commits/e824da5a2b33168fa4ef93c83295c8205acfadb7):

```
=== RUN   TestTransferLeaseToLaggingNode
--- FAIL: TestTransferLeaseToLaggingNode (0.05s)
    test_log_scope.go:107: can't use TestLogScope with secondary loggers active

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=TestTransferLeaseToLaggingNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTransferLeaseToLaggingNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

kv/kvserver.TestTransferLeaseToLaggingNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3945277&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3945277&tab=artifacts#/) on master @ [3f95a4bd83cce2952a12497de82692866b4da659](https://github.com/cockroachdb/cockroach/commits/3f95a4bd83cce2952a12497de82692866b4da659):


```
=== RUN   TestTransferLeaseToLaggingNode
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode174566069
    test_log_scope.go:80: use -show-logs to present logs inline
    replicate_queue_test.go:1218: condition failed to evaluate within 45s: remote is not making progress: 0
        goroutine 111778255 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x56cadd8, 0xc015bde000}, 0xc006dd18c0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x56cadd8, 0xc015bde000}, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestTransferLeaseToLaggingNode(0xc015bde000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:1218 +0xd37
        testing.tRunner(0xc015bde000, 0x47d1328)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTransferLeaseToLaggingNode174566069
--- FAIL: TestTransferLeaseToLaggingNode (78.36s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransferLeaseToLaggingNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestTransferLeaseToLaggingNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6261953?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6261953?buildTab=artifacts#/) on master @ [770ff3c545a51752490403da64d56fb397f49c5e](https://github.com/cockroachdb/cockroach/commits/770ff3c545a51752490403da64d56fb397f49c5e):


```
=== RUN   TestTransferLeaseToLaggingNode
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestTransferLeaseToLaggingNode2829826257
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:1796: condition failed to evaluate within 45s: remote is not making progress: 0
    panic.go:500: -- test log scope end --
--- FAIL: TestTransferLeaseToLaggingNode (98.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransferLeaseToLaggingNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestTransferLeaseToLaggingNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8241347?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8241347?buildTab=artifacts#/) on master @ [27ebfa830ac7122fc3b286136158fef615b20b8f](https://github.com/cockroachdb/cockroach/commits/27ebfa830ac7122fc3b286136158fef615b20b8f):


```
=== RUN   TestTransferLeaseToLaggingNode
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestTransferLeaseToLaggingNode1021163098
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:1965: condition failed to evaluate within 45s: remote is not making progress: 0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestTransferLeaseToLaggingNode1021163098
--- FAIL: TestTransferLeaseToLaggingNode (82.13s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransferLeaseToLaggingNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestTransferLeaseToLaggingNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8934103?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8934103?buildTab=artifacts#/) on master @ [14b43be03c1c246765be17aa15245493a42c401d](https://github.com/cockroachdb/cockroach/commits/14b43be03c1c246765be17aa15245493a42c401d):


```
=== RUN   TestTransferLeaseToLaggingNode
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestTransferLeaseToLaggingNode3293300275
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:1992: condition failed to evaluate within 45s: remote is not making progress: 0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestTransferLeaseToLaggingNode3293300275
--- FAIL: TestTransferLeaseToLaggingNode (96.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransferLeaseToLaggingNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -1908,6 +1908,8 @@ func TestTransferLeaseToLaggingNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t, "takes >1min under race")
+	skip.UnderDeadlock(t, "takes >1min under deadlock")
+	skip.UnderStressWithIssue(t, 53875)
 
 	ctx := context.Background()
 	clusterArgs := base.TestClusterArgs{

